site/src/components/introduction/building-a-chart.md

```diff
@@ -0,0 +1,188 @@
+
+But before we start putting this chart together, let's look at charting libraries in a more general sense ...
+
+## Charting APIs
+
+Charts are complex UI components, they are designed to be highly flexible in order to support a wide range of different chart types, styles and configurations. However, this flexibility is most often presented as a complex API of hundreds of classes and thousands of methods. Searching for the correct configuration that will yield the result you are after is a challenge, and you are heavily reliant on the designer of the chart having considered your use-case to make that configuration possible.
+
+D3 offers a fresh perspective, offering a different way to create charts and visualisations. It is an expressive, yet low-level tool, that if wielded correctly, can be used to create practically any visual representation you like.
+
+There are a number of charting libraries built with D3, however these invariably hide the underlying power of D3 behind the same-old complex and confusing APIs.
+
+With d3fc our aim is to enhance D3 by providing a suite of powerful charting components that make it easy for you to assemble exactly the chart you want. These components 'speak the language' of D3, supporting selections and datajoins. Furthermore, using the decorate pattern, you can enhance and manipulate the way in which they are rendered.
+
+With d3fc components you can quickly assemble bespoke charts, without hiding the power of D3.
+
+## Creating the data
+
+Typically chart data will be supplied by some external source, in JSON or CSV format. For the purposes of this example we'll just generate some random data. d3fc has a number of `data` utilities, including components that help construct realistic ransom data. For this example we'll use the [random walk component](/components/data/walk.html) to generate 12 datapoints, mapping them to create datapoints with `sales` and `month` properties:
+
+```js
+{{{creating-some-data-js}}}
+```
+
+Here's the resulting data:
+
+{{{ dynamic-include 'codepen' html="creating-some-data-html" js="creating-some-data-js"}}}
+{{{ creating-some-data-html }}}
+{{{ dynamic-include 'javascript' js="creating-some-data-js" }}}
+
+## Rendering the data
+
+The sales data is rendered as a bar chart. With D3 this typically involves rendering paths or rectangles and doing a bit of maths to locate each bar correctly, as illustrated in [Mike Bostock's excellent tutorial](http://bost.ocks.org/mike/bar/).
+
+d3fc has a [bar series component](/components/series/bar.html) (among others) that makes it much easier to render your data. To render the data, we create a bar series component and supply it with suitable D3 scales, configure the `xValue` and `yValue` accessors (which the component uses to extract the x and y values from the datapoints).
+
+The bar series is a regular D3 component, and is rendered via the `call` method on a D3 selection:
+
+```js
+{{{ codeblock rendering-a-series-js }}}
```

Comment:
This is where the xValue/yScale syntax is first introduced. There's a rebinding explanation later down in the 'tidying the chart' section, perhaps it could be moved up here?